Classes/Overrides/TYPO3/CMS/Backend/Controller/Page/TreeController.php

```diff
@@ -104,15 +104,13 @@ protected function pagesToFlatArray(array $page, int $entryPoint, int $depth = 0
             'siblingsCount' => $page['siblingsCount'] ?? 1,
             'siblingsPosition' => $page['siblingsPosition'] ?? 1,
             'allowDelete' => $backendUser->doesUserHaveAccess($page, Permission::PAGE_DELETE),
-            'allowEdit' => $backendUser->doesUserHaveAccess($page, Permission::PAGE_EDIT)
-                && $backendUser->check('tables_modify', 'pages')
-                && $backendUser->checkLanguageAccess(0),
+            'allowEdit' => $this->userHasAccessToModifyPagesAndToDefaultLanguage && $backendUser->doesUserHaveAccess($page, Permission::PAGE_EDIT),
         ];
 
-        if (!empty($page['_children']) || $this->getPageTreeRepository()->hasChildren($pageId)) {
+        if (!empty($page['_children']) || $this->getPageTreeRepository->hasChildren($pageId)) {
```

Comment:
I think the property needs to be $this->pageTreeRepository not $this->getPageTreeRepository

Reply:
Yes you are totally right. In my local patch I also use the $this->pageTreeRepository property. Thanks for the hint.